packages/components/utils/src/useKeyboard/useKeyboard.ts

```diff
@@ -0,0 +1,55 @@
+import { useState, useEffect, useCallback } from 'react';
+
+export interface UseKeyboardProps {
+  /**
+   * @description defines which element is the event attached to
+   * @default document
+   */
+  ref?: Document | Window | HTMLElement;
```

Comment:
I wouldn't call it `ref`, since it can be confused with `ReactRef`. I'd rather call it just an element.

Reply:
What if I want to pass `ref.current` which can be `null`? 
```tsx
const myRef = React.createRef(null)
```
it is the most common pattern.
In this case I won't be able to use this hook, cause I can't pass `null` as an argument. And I can't wrap hooks into conditionals.

Reply:
I went with passing the full `ref` object as the hook needs to monitor current when it changes from `null` to defined. 

That way the consumer wouldn't need to fiddle much with it, and by default the event is attached to `document`